src/World.cpp

```diff
@@ -1130,19 +1130,36 @@ void cWorld::TickMobs(std::chrono::milliseconds a_Dt)
 		}  // for i - AllFamilies[]
 	}  // if (Spawning enabled)
 
-	// move close mobs
-	cMobProximityCounter::sIterablePair allCloseEnoughToMoveMobs = MobCensus.GetProximityCounter().getMobWithinThosesDistances(-1, 64 * 16);// MG TODO : deal with this magic number (the 16 is the size of a block)
-	for (cMobProximityCounter::tDistanceToMonster::const_iterator itr = allCloseEnoughToMoveMobs.m_Begin; itr != allCloseEnoughToMoveMobs.m_End; ++itr)
+	class cCallback : public cEntityCallback
 	{
-		itr->second.m_Monster.Tick(a_Dt, itr->second.m_Chunk);
-	}
+		virtual bool Item(cEntity * a_Entity) override
+		{
+			if (!a_Entity->IsMob())
+			{
+				return false;
+			}
+
+			cMonster * Monster = static_cast<cMonster*>(a_Entity);
+
+			// Tick close mobs
+			if (Monster->GetParentChunk()->HasAnyClients())
```

Comment:
That depends on which version of my commit this is, I don't remember for sure anymore.
I want to thoroughly review all cases for Get/SetParentChunk uses before merging #2986 , I'll change this accordingly, if needed, once done.